rewrite-gradle/src/main/java/org/openrewrite/gradle/AddGradleWrapper.java

```diff
@@ -0,0 +1,152 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Parser;
+import org.openrewrite.Recipe;
+import org.openrewrite.SourceFile;
+import org.openrewrite.binary.Binary;
+import org.openrewrite.binary.BinaryParser;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.properties.PropertiesParser;
+import org.openrewrite.properties.tree.Properties;
+import org.openrewrite.text.PlainText;
+import org.openrewrite.text.PlainTextParser;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.openrewrite.PathUtils.equalIgnoringSeparators;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddGradleWrapper extends Recipe {
+    private static final String DEFAULT_VERSION = "7.4.2";
+    private static final String DEFAULT_DISTRIBUTION = "bin";
+    private static final Path WRAPPER_JAR_LOCATION = Paths.get("gradle/wrapper/gradle-wrapper.jar");
+    private static final Path WRAPPER_PROPERTIES_LOCATION = Paths.get("gradle/wrapper/gradle-wrapper.properties");
+    private static final Path WRAPPER_SCRIPT_LOCATION = Paths.get("gradlew");
+    private static final Path WRAPPER_BATCH_LOCATION = Paths.get("gradlew.bat");
+
+    @Override
+    public String getDisplayName() {
+        return "Add Gradle wrapper";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add a Gradle wrapper where one does not exist.";
+    }
+
+    @Option(displayName = "Version",
+            description = "The version of Gradle to use. Defaults to " + DEFAULT_VERSION,
+            example = DEFAULT_VERSION,
+            required = false
+    )
+    @Nullable
+    String version;
+
+    @Option(displayName = "Distribution type",
+            description = "The distribution of Gradle to use. \"bin\" includes Gradle binaries. " +
+                    "\"all\" includes Gradle binaries, source code, and documentation. " +
+                    "Defaults to \"bin\".",
+            example = "bin",
+            required = false
+    )
+    @Nullable
+    String distribution;
+
+    @Option(displayName = "Distribution Url",
+            description = "The URL to download the Gradle distribution from.",
+            example = "https://services.gradle.org/distributions/gradle-" + DEFAULT_VERSION + "-" + DEFAULT_DISTRIBUTION + ".zip",
+            required = false
+    )
+    @Nullable
+    String distributionUrl;
```

Comment:
We should mention in the description of these parameters that if `distributionUrl` is provided that `version` and `distribution` will be ignored.

Reply:
Fixed via [daa7aef](https://github.com/openrewrite/rewrite/pull/1560/commits/daa7aefbe51144a7a8458b39a5c2593c3f5d2a93)